plugins/outputs/postgresql/postgresql.go

```diff
@@ -2,27 +2,32 @@ package postgresql
 
 import (
 	"log"
-	"sort"
+	"sync"
 
 	"github.com/influxdata/telegraf"
 	"github.com/influxdata/telegraf/plugins/outputs"
-)
-
-const (
-	tagsJSONColumn   = "tags"
-	fieldsJSONColumn = "fields"
+	"github.com/influxdata/telegraf/plugins/outputs/postgresql/columns"
+	"github.com/influxdata/telegraf/plugins/outputs/postgresql/db"
+	"github.com/influxdata/telegraf/plugins/outputs/postgresql/tables"
+	"github.com/influxdata/telegraf/plugins/outputs/postgresql/utils"
 )
 
 type Postgresql struct {
-	db                dbWrapper
-	Address           string
-	Schema            string
-	TagsAsForeignkeys bool
-	TagsAsJsonb       bool
-	FieldsAsJsonb     bool
-	TableTemplate     string
-	TagTableSuffix    string
-	tables            tableKeeper
+	db                          db.Wrapper
+	Connection                  string
+	Schema                      string
+	DoSchemaUpdates             bool
+	TagsAsForeignkeys           bool
+	CachedTagsetsPerMeasurement int
+	TagsAsJsonb                 bool
+	FieldsAsJsonb               bool
+	TableTemplate               string
+	TagTableSuffix              string
+	tables                      tables.Manager
+	tagCache                    tagsCache
+	rows                        transformer
+	columns                     columns.Mapper
+	dbConnLock                  *sync.Mutex
```

Comment:
It's conventional to group the mutex above the fields it is intended to protect. It would be useful to add a comment to say which fields must be accessed only when holding the mutex lock.

Also the mutex would be better as a value not a pointer. Using a pointer means copies of the `Postgresql` type will share the same mutex which would be a bad situation.

Reply:
@blagojts, thank you very much.

> About `sslcompression`, since it's not considered safe, it is not supported by the tls implementation in the `crypto/tls` go package.

Oh thats unfortunate... I dont care about security in this case. sslcompression is (afaik) the only way to enable compression for the tcp connection. Thanks for the info.